enterprise/server/atime_updater/atime_updater.go

```diff
@@ -162,13 +175,47 @@ func (u *atimeUpdater) Enqueue(ctx context.Context, instanceName string, digests
 	}
 
 	// Now merge the new digests into the update.
-	for key := range keys {
+	enqueued := 0
+	duplicate := 0
+	dropped := 0
+	logWarning := true
+	for key, count := range keys {
 		if len(pendingUpdate.digests) >= u.maxDigestsPerUpdate {
-			log.CtxInfof(ctx, "FindMissingBlobsRequest for updating remote atime for group %s too large, dropping some pending atime updates", groupID)
-			break
+			if logWarning {
+				log.CtxInfof(ctx, "FindMissingBlobsRequest for updating remote atime for group %s too large, dropping some pending atime updates", groupID)
+				logWarning = false
+			}
+			dropped += count
+			continue
+		}
+		if _, ok := pendingUpdate.digests[key]; ok {
+			duplicate += count
+			continue
 		}
 		pendingUpdate.digests[key] = struct{}{}
+		enqueued++
+		duplicate += count - 1
```

Comment:
why is this considered a duplicate?

Reply:
The first one is enqueued and any others are considered duplicates. Agree that it's slightly odd that the "too_many_batches" above is recorded once per instance, but I figured that measuring duplicates as those that actually make it into the queue was reasonable. We can always change it if we need a slightly different definition in the future.